File: src/commands/project-config.ts
import { safeInvoke } from '~/utils/invoke'

import type { ProjectConfig } from '~/types/project-config'

function readProjectConfig(projectPath: string): Promise<ProjectConfig> {
  return safeInvoke('read_project_config_cmd', { projectPath })
}

function writeProjectConfig(projectPath: string, config: ProjectConfig): Promise<void> {
  return safeInvoke('write_project_config_cmd', { projectPath, config })
}

export const projectConfigCmds = {
  readProjectConfig,
  writeProjectConfig,
}

File: src-tauri/src/commands/project_config.rs
use std::path::Path;

use crate::vfs::{read_project_config, write_project_config, ProjectConfig};

use super::AppResult;

#[tauri::command]
pub fn read_project_config_cmd(project_path: String) -> AppResult<ProjectConfig> {
    read_project_config(Path::new(&project_path))
}

#[tauri::command]
pub async fn write_project_config_cmd(
    project_path: String,
    config: ProjectConfig,
) -> AppResult<()> {
    write_project_config(Path::new(&project_path), &config).await
}
